packages/ngtools/webpack/src/ngcc_processor.ts

```diff
@@ -49,6 +50,51 @@ export class NgccProcessor {
     }
   }
 
+  async process() {
+    const timeLabel = 'NgccProcessor.process';
+    time(timeLabel);
+
+    const corePackage = this.tryResolvePackage('@angular/core', this._nodeModulesDirectory);
+
+    // If the package.json is read only we should skip calling NGCC.
+    // With Bazel when running under sandbox the filesystem is read-only.
+    if (corePackage && isReadOnlyFile(corePackage)) {
+      timeEnd(timeLabel);
+
+      return;
+    }
+
+    // The below is a workaround for NGCC fork clustering issues.
+    // The current NGCC implemementation of parallelism using clusting is problematic when using via an API.
+    // because when using cluster.fork the entire process life cycle will get restarted. In a nutshell what happens
+    // is that we do `ng build` and NGCC does `cluster.fork()` for each fork `ng build` is re run.
+    // See: https://github.com/angular/angular/issues/35717
+    setupMaster({
+      exec: require.resolve('@angular/compiler-cli/ngcc/main-ngcc.js'),
+      args: [
+        '--source', /** basePath */
+        this._nodeModulesDirectory,
+        '--properties', /** propertiesToConsider */
+        ...this.propertiesToConsider,
+        '--first-only', /** compileAllFormats */
+        '--create-ivy-entry-points', /** createNewEntryPointFormats */
+        '--async',
+      ],
+    });
+
+    await mainNgcc({
+      basePath: this._nodeModulesDirectory,
+      propertiesToConsider: this.propertiesToConsider,
+      compileAllFormats: false,
+      createNewEntryPointFormats: true,
+      logger: this._logger,
+      async: true,
+      pathMappings: this._pathMappings,
+    });
```

Comment:
I think we will also need to run ngcc again against `basePath`s that have been computed from the `pathMappings` property to catch local libraries too.

Reply:
How would that work, without providing a module name?


Currently for those will be handled via the sync pipeline.

Reply:
Actually don't worry. It is handled inside ngcc already:  https://github.com/angular/angular/blob/a581773887eaf872f484e4a6469ad1a6068d8305/packages/compiler-cli/ngcc/src/entry_point_finder/directory_walker_entry_point_finder.ts#L32-L34